tests/console/kubeadm.pm

```diff
@@ -19,24 +19,20 @@ use utils "systemctl";
 sub run {
     select_console("root-console");
 
-    record_info 'etcd', 'Stop etcd and clean up';
-    systemctl 'disable --now etcd';
-    script_run 'rm -r /var/lib/etcd/*';
-
-    record_info 'Setup', 'Test: Package Installation';
-    my $packages = 'kubernetes-client kubernetes-kubelet kubernetes-kubeadm docker-kubic cri-tools';
-    trup_install($packages);
+    record_info 'Test #1', 'Test: Initialize kubeadm';
+    assert_script_run('kubeadm init --cri-socket=/var/run/crio/crio.sock --pod-network-cidr=10.244.0.0/16', 180);
 
-    record_info 'Prepare', 'Test: Enable and start required services';
-    systemctl('enable --now kubelet docker');
-    systemctl('is-active docker');
-    systemctl('is-active kubelet');
+    record_info 'Test #2', 'Test: Configure kubectl';
+    assert_script_run('mkdir -p ~/.kube');
+    assert_script_run('cp -i /etc/kubernetes/admin.conf ~/.kube/config');
 
-    record_info 'Test #1', 'Test: Initialize kubeadm';
-    record_soft_failure "bsc#1093132" if script_run('kubeadm init');
-    script_run('kubeadm reset');
-    assert_script_run('kubeadm init', 180);
+    record_info 'Test #3', 'Test: Configure flannel';
+    assert_script_run('kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/bc79dd1505b0c8681ece4de4c0d86c5cd2643275/Documentation/kube-flannel.yml', 180);
```

Comment:
I think we should avoid external sources.

- Security
- Test reliability
- False positives

I can think at the moment two and a half possible solutions:

1. Putting the file as an assets here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/data/caasp
2.1. Packaging it in https://build.opensuse.org/project/show/devel:openSUSE:QA
2.2. Or even packaging into factory (would it make sense to provide this file to users as a package?)

Reply:
No way, no how, sorry, the Kubernetes way of doing things requires external sources, and we need to test the way users are going to use the cluster.

Sorry, I know it's not ideal, but I will not accept this test becoming artificial